### PR TITLE
Improvements to compiled mode

### PR DESCRIPTION
**Pull request by maleadt**
cc @DilumAluthge

### CODE REVIEW

File: src/jobs/PkgEvalJob.jl [hunk 1]
```diff
@@ -108,13 +108,16 @@ function PkgEvalJob(submission::JobSubmission)
...
     if haskey(submission.kwargs, :compiled)
-        val = Meta.parse(submission.kwargs[:compiled])
-        if !isa(val, Bool)
-            error("invalid argument to `compiled` keyword")
+        expr = Meta.parse(submission.kwargs[:compiled])
+        if !isa(expr, QuoteNode)
+            error("invalid argument to `compiled` keyword (should be a Symbol)")
+        end
+        compiled = expr.value
+        if !in(compiled, [:none, :primary, :against, :both])
```
Comment (DilumAluthge) [inline src/jobs/PkgEvalJob.jl:116]: Could be nice to have a comment that explains the difference between these four options.

Comment (maleadt) [reply src/jobs/PkgEvalJob.jl:116]: There's an according change in the README. Or do you mean details on the values (which I assumed to be fairly self-explanatory, but I can add some more text).

**Comment by maleadt** [comment]
Nanosoldier is currently doing its daily evaluation, so I'll test and deploy this tomorrow.

File: src/jobs/PkgEvalJob.jl (hunk 1 above)
Comment (maleadt) [reply src/jobs/PkgEvalJob.jl:116]: That comparison does take into account build flags and compilation mode: https://github.com/JuliaCI/Nanosoldier.jl/blob/eff19637c464e02af520f7eef6e181418472cc1a/src/jobs/PkgEvalJob.jl#L368-L370

So if any of those differ, an identical SHA will be accepted and a comparison build performed. So it's just `runtests(..., vs="...", compiled=:primary)`

Comment (DilumAluthge) [reply src/jobs/PkgEvalJob.jl:116]: So, suppose that I comment the following on a specific commit:

```
@nanosoldier runtests(ALL, vs="FOO", compiled=:primary)
```

What do I put in place of `FOO`, to tell Nanosoldier that both the primary and the against should use the same commit hash (i.e. the commit that I post the comment on)?

Comment (maleadt) [reply src/jobs/PkgEvalJob.jl:116]: There isn't a special option for using the same. Just specify the commit or the branch.

**Pull request merged**